.github/workflows/integration.yml

```diff
@@ -0,0 +1,68 @@
+name: integration
```

Comment:
I am more familiar with the GitHub Actions syntax. Most of the content is also just a variation of the one we have in [react-native-test-app](https://github.com/microsoft/react-native-test-app/blob/trunk/.github/workflows/build.yml).

Reply:
All our other validation happens on ADO so that'd be more consistent. If it's not too much trouble I think that'd make more sense just to keep in line with everything else

Reply:
I had a look at [`apple-pr.yml`](https://github.com/microsoft/react-native-macos/blob/master/.ado/apple-pr.yml), assuming this is where I would add the integration job, and noticed that it's supposed to run on main and stable branches, but it doesn't look like it currently is?

![image](https://user-images.githubusercontent.com/4123478/140488603-0b431371-8b95-41a7-9432-1afa99530c0a.png)

I think it would make sense for this pipeline to also run when things land on master and stable. How much work would it be to get that set up in ADO?

Reply:
If you copy the props from pr: to trigger: then the apple-pr.yml will run when things land too.  Although that should generally just be the same as what was just run on the PR. -- But you can go ahead and make that change if you need it.

Reply:
Migrated to Azure Pipelines. Of course, now I don't know whether this works until I create a pipeline in ADO after this merges. I don't suppose you know if it's possible to test this earlier than that?

Reply:
Not easily, in my experience. I've had to basically authenticate my fork to run pipelines on the same ADO instance to test beforehand, or (with admin privileges) manually run a pipeline off a specific branch of the repo. Neither options were fun, and prone to failure. If someone else knows a better way I'm all ears. It sounds like Github actions however, lets you validate your pipeline in the PR that creates it since there's no extra setup. That's a good plus on the github actions side

Reply:
You can test the pipeline here:
https://dev.azure.com/ms/react-native/_apps/hub/ms.vss-build-web.ci-designer-hub?pipelineId=531&branch=tido%2Fadd-integration-pipeline&nonce=chhjXQeiLIWrlJkyG6Qqrg%3D%3D

Reply:
Thanks, I got a "successful" build with the latest changes: https://dev.azure.com/ms/react-native/_build/results?buildId=240874&view=logs